core/src/main/java/hudson/PluginWrapper.java

```diff
@@ -530,6 +564,95 @@ public void disable() throws IOException {
         }
     }
 
+    /**
+     * Disable this plugin using a strategy.
+     * @param strategy strategy to use
+     * @return an object representing the result of the disablement of this plugin and its dependants plugins.
+     * @throws IOException If this plugin or one dependant plugin raise this exception during its disablement. It could
+     * be raised because the disable file cannot be created.
+     */
+    @Nonnull
+    public PluginDisableResult disable(@Nonnull PluginDisableStrategy strategy) throws IOException {
```

Comment:
nit: for the `@Nonnull` you put it once before the plugin, other time between public and the return typed. I would prefer the latter but at least keep consistency.